src/rgw/rgw_common.h

```diff
@@ -263,13 +263,13 @@ enum {
 
  /* size should be the required string size + 1 */
 extern int gen_rand_base64(CephContext *cct, char *dest, int size);
-extern int gen_rand_alphanumeric(CephContext *cct, char *dest, int size);
-extern int gen_rand_alphanumeric_lower(CephContext *cct, char *dest, int size);
-extern int gen_rand_alphanumeric_upper(CephContext *cct, char *dest, int size);
-extern int gen_rand_alphanumeric_no_underscore(CephContext *cct, char *dest, int size);
-extern int gen_rand_alphanumeric_plain(CephContext *cct, char *dest, int size);
+extern void gen_rand_alphanumeric(CephContext *cct, char *dest, int size);
+extern void gen_rand_alphanumeric_lower(CephContext *cct, char *dest, int size);
+extern void gen_rand_alphanumeric_upper(CephContext *cct, char *dest, int size);
+extern void gen_rand_alphanumeric_no_underscore(CephContext *cct, char *dest, int size);
+extern void gen_rand_alphanumeric_plain(CephContext *cct, char *dest, int size);
 
```

Comment:
Not directly related to your patch, but we should really kill 'extern' on function prototypes when we're changing them. Function prototypes are extern, there's no reason to add it, it just adds noise.